--- hunspell-1.2.1/configure.ac.orig	2007-11-01 19:23:22.000000000 +0100
+++ hunspell-1.2.1/configure.ac	2008-01-03 18:23:21.059100740 +0100
@@ -59,13 +59,13 @@
 )
 AS_IF(
    [test "x$with_ui" != xno],
-   [AC_CHECK_LIB(ncursesw,tparm,CURSESLIB=-lncursesw,
-     AC_CHECK_LIB(curses,tparm,CURSESLIB=-lcurses,
-       AC_CHECK_LIB(ncurses,tparm,CURSESLIB=-lncurses)))
+   [AC_CHECK_LIB(ncursesw,tparm,CURSESLIB="-lncursesw -ltinfow",
+     AC_CHECK_LIB(curses,tparm,CURSESLIB="-lcurses -ltinfo",
+       AC_CHECK_LIB(ncurses,tparm,CURSESLIB="-lncurses -ltinfo")))
    if test "$CURSESLIB" != "" ; then
      echo Compiling with curses user interface.
      AC_DEFINE(HAVE_CURSES_H,1,"Define if you have the <curses.h> header")
-     if test "$CURSESLIB" != "-lncursesw" ; then
+     if test "$CURSESLIB" != "-lncursesw -ltinfow" ; then
        echo "No Unicode support on interactive console. (Install Ncursesw library.)"
      else
        AC_DEFINE(HAVE_NCURSESW_H,1,"Define if you have the <ncursesw/curses.h> header")       
@@ -83,7 +83,7 @@
 )
 rl=n
 AS_IF([test "x$with_readline" != xno],
-   [AC_CHECK_LIB(curses,tparm,TERMLIB=-lncurses,
+   [AC_CHECK_LIB(tinfo,tparm,TERMLIB=-ltinfo,
      AC_CHECK_LIB(termcap,tgetent,TERMLIB=-ltermcap))
    LDSAVE=$LDFLAGS
    LDFLAGS="$LDFLAGS $TERMLIB"

